.github/workflows/docs.yml

```diff
@@ -0,0 +1,35 @@
+name: Documentation
+
+on:
+  pull_request:
+  push:
+  workflow_dispatch:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          pip install -r docs/doc-requirements.txt
+
+      - name: Build docs
+        run: |
+          cd docs
+          make html
+
+      - name: publish to github pages
+        if: github.ref == 'refs/heads/main'
+        run: |
+            pip install ghp-import
+            ghp-import --no-jekyll --push --message "Update documentation [skip ci]" docs/_build/html
```

Comment:
I wonder if we will have a github user configured to indicate this is a bot, otherwise the git user will be of the person that triggered the workflow - which would be whoever pushed / merged a PR i think. That is probably acceptable as well though.

Reply:
It seems like they have no configuration about setting the git user to make the commit, so I assume its done with the current user without modification of that etc.

https://pypi.org/project/ghp-import/

In z2jh we have done the following: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/04de3447397bb85b495d0302984d36c61bc75da5/.github/workflows/vuln-scan.yaml#L204